app/src/components/Leverager/index.js

```diff
@@ -1,79 +1,371 @@
-    // }
-  };
-  handleFloorChange = ({ target: { value = "" } }) => {
-    const priceFloor = parseFloat(value) || null;
+    initWeb3(web3)
+    if (!web3.currentProvider.isMetaMask === true) {
+      this.setState({ error: 'Please install metamask' })
+    }
+  }
+  handleFloorChange = ({ target: { value = '' } }) => {
+    const priceFloor = parseFloat(value) || null
     if (priceFloor > this.state.ETHUSD && !this.state.error) {
       this.setState({
-        error: `Price floor must be below current ETH price ($${
-          this.state.ETHUSD
-        })`
-      });
-      this.setState({ priceFloor });
+        error: `Price floor must be below current ETH price ($${this.state.ETHUSD})`
+      })
+      this.setState({ priceFloor })
     } else {
-      !!this.state.error && this.setState({ error: false });
-      this.calculateRatioAndLeverage(priceFloor);
-      this.setState({ priceFloor });
+      !!this.state.error && this.setState({ error: false })
+      this.calculateRatioAndLeverage(priceFloor)
+      this.setState({ priceFloor })
     }
-  };
-  handleETHChange = ({ target: { value = "" } }) => {
-    const AmtETH = parseFloat(value) || null;
-    this.setState({ AmtETH });
-  };
+  }
+  handleETHChange = ({ target: { value = '' } }) => {
+    const AmtETH = parseFloat(value) || null
+    this.setState({ AmtETH })
+  }
   calculateRatioAndLeverage = priceFloor => {
-    const collatRatio = this.calculateCollatRatio(priceFloor);
-    const leverage = this.calculateLeverage(collatRatio);
-    this.setState({ collatRatio, leverage });
-  };
+    const collatRatio = this.calculateCollatRatio(priceFloor)
+    const leverage = this.calculateLeverage(collatRatio)
+    this.setState({ collatRatio, leverage })
+  }
   calculateCollatRatio = priceFloor => {
-    const CollatRatio = this.state.ETHUSD * 1.5 / priceFloor;
-    return twoDecimalFloat(CollatRatio);
-  };
-  setDepth = ({ target: { value = "" } }) => {
-    const depth = parseInt(value, 10) || null;
+    const CollatRatio = this.state.ETHUSD * 1.5 / priceFloor
+    return twoDecimalFloat(CollatRatio)
+  }
+  setDepth = ({ target: { value = '' } }) => {
+    const depth = parseInt(value, 10) || null
     this.setState({ depth }, () => {
-      this.calculateRatioAndLeverage(this.state.priceFloor);
-    });
-  };
+      this.calculateRatioAndLeverage(this.state.priceFloor)
+    })
+  }
   calculateLeverage = collatRatio => {
-    const steps = parseInt(this.state.depth, 10) || 0;
+    const steps = parseInt(this.state.depth, 10) || 0
     let leverage = 1,
-      currStep = 1;
+      currStep = 1
     for (let i = 0; i < steps; i++) {
-      currStep = currStep * 1 / collatRatio;
-      leverage += currStep;
+      currStep = currStep * 1 / collatRatio
+      leverage += currStep
     }
-    return twoDecimalFloat(leverage);
-  };
+    return twoDecimalFloat(leverage)
+  }
   createPosition = () => {
+    const leveragerProxy = new web3.eth.Contract(
```

Comment:
We shouldn't hardcode the ABI here. Can we pull it in from the builds/ folder instead?

Reply:
Looks like we need to have the ABIs stored within the src/ folder or else React complains